[Regression] Bulk file actions broken on Nextcloud 32 since richdocuments 9.2.0 (markdown action uses NC 33 API)

**Describe the bug**

Since updating the app to **9.2.0**, the multi-select action bar in the Files app is broken on **Nextcloud 32**. Selecting several files and trying a bulk action (Copy, Move, Delete, …) no longer works — the batch actions do not run. Acting on a single file still works.

The reason is a file action registered by richdocuments whose `enabled()` callback throws under Nextcloud 32.

The Files app builds the multi-select action bar by evaluating `enabled()` of **every** registered file action (`FilesListTableHeaderActions.vue`, computed `enabledFileActions`). That filter has no per-action `try/catch`, so a single throwing `enabled()` takes down the whole batch bar. The single-row path (`FileEntryMixin.ts`) does catch per action, which is why single-file actions still work.

The throwing action is `office-open-markdown` in `src/file-actions.js`:

```js
enabled: ({ nodes }) => {
    if (nodes.length !== 1) {   // file-actions.js:70 → nodes is undefined on NC 32
```

`enabled()` destructures `nodes` from a **context object** — that is the Nextcloud **33+** file-actions API. On **Nextcloud 32** the callback still receives the **node array** directly, so `nodes` is `undefined` and `nodes.length` throws.

The same file proves the inconsistency: the older `office-open-pdf` action in the same file uses the NC 32 array API and works fine:

```js
enabled: (files) => {
    if (files.length !== 1) { ... }   // array API → OK on NC 32
```

So the NC-33 API code was shipped to NC 32 without being adapted.

**To Reproduce**
Steps to reproduce the behavior:
1. On a Nextcloud 32 instance with richdocuments 9.2.0, open the Files app.
2. Tick the checkboxes of two or more files.
3. Try any bulk action from the top action bar (e.g. Move or copy).
4. Nothing happens; the browser console shows `TypeError: Cannot read properties of undefined (reading 'length') at Object.enabled (file-actions.js:70)`.

**Expected behavior**
Bulk actions on multiple selected files work as before 9.2.0. The richdocuments file action must not throw on Nextcloud 32.

**Screenshots**
n/a

**Client details:**
 - OS: Linux (desktop)
 - Browser: Chromium / Firefox (any)
 - Device: desktop

## Server details

**Operating system**: Ubuntu

**Web server:** Apache/2.4.68

**Database:** MySQL / MariaDB

**PHP version:** 8.4.23

**Nextcloud version:** 32.0.13.1

**Version of the richdocuments app**: 9.2.0

**Version of Collabora Online**: not relevant — this is a frontend-only JavaScript API mismatch, independent of the COOL server version.

**Configuration of the richdocuments app**
```
(available on request — not relevant to this frontend bug)
```

<details>
<summary>Logs</summary>

#### Browser log
```
FileEntryMixin.ts:211 [ERROR] files: Error while checking action
  error: TypeError: Cannot read properties of undefined (reading 'length')
      at Object.enabled

selection.ts:15  Uncaught TypeError: Cannot read properties of undefined (reading 'length')
    at Object.enabled (file-actions.js:70:13)
    at FilesListTableHeaderActions.vue:74:61
    at Array.filter (<anonymous>)
    at o.enabledFileActions (FilesListTableHeaderActions.vue:74:18)
```

</details>

---

## Where the mistake was made

The `office-open-markdown` action was added on `main` (for NC 33+) in PR #5873 and then **backported to `stable32`** in PR #5918, shipped as **9.2.0**. The `enabled: ({ nodes })` context-object signature is correct for NC 33+, but on the `stable32` branch it needed the NC 32 array signature (as `office-open-pdf` in the same file already uses). The backport carried the NC-33 API onto NC 32 unchanged.

- Introduced: PR #5873 (main), commit `7ec1f6c6`, 2026-07-17
- Backported: PR #5918 (`[stable32]`), commit `86ef4507`, merged 2026-07-24 → **first affected release: richdocuments 9.2.0**
- 9.1.1 did not have this action, which is why bulk actions worked before.

Confirmed by isolation: disabling the `tables` app (which also ships a `file-actions.js`) changed nothing; disabling `richdocuments` restored bulk actions immediately.

--- a/src/file-actions.js
+++ b/src/file-actions.js
@@ -66,24 +66,22 @@ const openMarkdown = {
 			{ productName: getCapabilities().productName })
 	},
 
-	enabled: ({ nodes }) => {
-		if (nodes.length !== 1) {
+	enabled: (files) => {
+		if (files.length !== 1) {
 			return false
 		}
 
-		if ((nodes[0].permissions & Permission.READ) === 0) {
+		if ((files[0].permissions & Permission.READ) === 0) {
 			return false
 		}
 
-		const isMarkdown = nodes[0].mime === 'text/markdown'
+		const isMarkdown = files[0].mime === 'text/markdown'
 		// Only enable the file action when Collabora is not the default handler
 		const optionalMimetypes = getCapabilities().mimetypesNoDefaultOpen
 		return isMarkdown && optionalMimetypes.includes('text/markdown')
 	},
 
-	exec: ({ nodes }) => {
-		const file = nodes[0]
-
+	exec: (file) => {
 		// If no viewer API, we can't open the document
 		if (!OCA.Viewer) {
 			return

